File: 2026/week1/sort_array_by_parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        vector<int> result(nums.size(), 0);
        int p1 = 0;
        int p2 = nums.size() - 1;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] % 2 == 0) {
                result[p1] = nums[i];
                p1++;
            } else {
                result[p2] = nums[i];
                p2--;
            }
        }
        return result;
    }
};

File: 2026/week1/sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        l = [0] * len(nums)
        p1 = 0
        p2 = len(l) - 1
        for i in nums:
            if p2 < p1:
                continue
            if i % 2 == 0:
                l[p1] = i
                p1 += 1
            else:
                l[p2] = i
                p2 -= 1
        return l

        